pkg/util/test/util.go

```diff
@@ -937,6 +918,83 @@ func ValidateStork(pxImageList map[string]string, cluster *corev1.StorageCluster
 	return nil
 }
 
+// ValidateAutopilot validates Autopilot components and images
+func ValidateAutopilot(pxImageList map[string]string, cluster *corev1.StorageCluster, timeout, interval time.Duration) error {
+	autopilotDp := &appsv1.Deployment{}
+	autopilotDp.Name = "autopilot"
+	autopilotDp.Namespace = cluster.Namespace
+	autopilotConfigMapName := "autopilot-config"
+
+	if cluster.Spec.Autopilot != nil && cluster.Spec.Autopilot.Enabled {
+		logrus.Debug("Autopilot is Enabled in StorageCluster")
+
+		// Validate autopilot deployment and pods
+		if err := validateDeployment(autopilotDp, timeout, interval); err != nil {
+			return err
+		}
+
+		var autopilotImageName string
+		if cluster.Spec.Autopilot.Image == "" {
+			if value, ok := pxImageList[autopilotDp.Name]; ok {
+				autopilotImageName = value
+			} else {
+				return fmt.Errorf("failed to find image for %s", autopilotDp.Name)
+			}
+		} else {
+			autopilotImageName = cluster.Spec.Autopilot.Image
+		}
+
+		autopilotImage := util.GetImageURN(cluster, autopilotImageName)
+		if err := validateImageOnPods(autopilotImage, cluster.Namespace, map[string]string{"name": autopilotDp.Name}); err != nil {
+			return err
+		}
+
+		// Validate Autopilot ClusterRole
+		_, err := rbacops.Instance().GetClusterRole(autopilotDp.Name)
+		if errors.IsNotFound(err) {
+			return fmt.Errorf("failed to validate ClusterRole %s, Err: %v", autopilotDp.Name, err)
+		}
+
+		// Validate Autopilot ClusterRoleBinding
+		_, err = rbacops.Instance().GetClusterRoleBinding(autopilotDp.Name)
+		if errors.IsNotFound(err) {
+			return fmt.Errorf("failed to validate ClusterRoleBinding %s, Err: %v", autopilotDp.Name, err)
+		}
+
+		// Validate Autopilot ConfigMap
+		_, err = coreops.Instance().GetConfigMap(autopilotConfigMapName, autopilotDp.Namespace)
```

Comment:
Also, validate the `autopilot` service account both here and in the else block.

Reply:
Added check for that